src/WattleScript.Interpreter/Execution/VM/Processor/Processor_InstructionLoop.cs

```diff
@@ -1229,7 +1229,7 @@ private int ExecAddStr(int instructionPtr)
 			DataType dvAType = dvA.Type;
 			DataType dvBType = dvB.Type;
 			
-			if (dvA.TryCastToNumber(out var rn) && dvB.TryCastToNumber(out var ln))
+			if (dvA.Type == DataType.Number && dvB.Type == DataType.Number && dvA.TryCastToNumber(out double rn) && dvB.TryCastToNumber(out double ln))
```

Comment:
See review comment on how to fix this

Reply:
Fixed in aa41897479167bc448ff499436c1ebfef522d654